File: Modules/SEDO/EMS/SIE.Web.EMS/Equipments/Models/Commands/SelModelCheckCommand.js
SIE.defineCommand('SIE.Web.EMS.Equipments.Models.Commands.SelModelCheckCommand', {
    extend: 'SIE.Web.EMS.Common.Commands.SelCheckProjectCommand',
    meta: { text: "选择", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },
    /**
     * canExecute 是否执行
     * @param {} view 当前视图
     * @returns {}
     */
    canExecute: function (view) {
        var entity = view.getParent().getCurrent();

        if (entity == null || entity.data == null) {
            //所属父对象（设备型号）为空 或 数据为空时，不能点击选择点检项目
            return false;
        }

        //新增时，不能点击选择校验项目
        if (entity.isNew()) {
            return false;
        }

        return true;
    },
});

File: Modules/SEDO/EMS/SIE.Web.EMS/Equipments/Models/Commands/SelModelCheckCommand.cs
using SIE.Domain;
using SIE.Domain.Validation;
using SIE.EMS.Common.Entity;
using SIE.EMS.Equipments.Models;
using SIE.EMS.MainenanceProjects;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;

namespace SIE.Web.EMS.Equipments.Models.Commands
{
    /// <summary>
    /// 添加点检项目
    /// </summary>
    [JsCommand("SIE.Web.EMS.Equipments.Models.Commands.SelModelCheckCommand")]
    public class SelModelCheckCommand : ViewCommand
    {
        /// <summary>
        /// 执行添加
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var meta = ClientEntities.Find(args.Type);
            var savedData = RF.Find(meta.EntityType).NewList();
            var checkProjectInfos = args.Data.ToJsonObject<List<CheckProjectInfo>>();
            Check.NotNullOrEmpty(checkProjectInfos, nameof(checkProjectInfos));
            if (null == checkProjectInfos || checkProjectInfos.Count == 0)
                throw new ValidationException("点检项目列表不能为空".L10N());
            var projectDetailIds = checkProjectInfos.Select(m => m.ProjectDetailId).ToList();
            var projectDetails = RT.Service.Resolve<ProjectDetailController>().GetProjectDetails(projectDetailIds);

            foreach (var item in checkProjectInfos)
            {
                var detail = projectDetails.FirstOrDefault(m => m.Id == item.ProjectDetailId);

                if (!detail.CycleType.HasValue)
                {
                    throw new ValidationException("点检项目【{0}】的周期类型为空".L10nFormat(detail.Name));
                }

                var checkProject = new EquipModelCheckProject();
                checkProject.EquipModelId = item.SourceId;
                checkProject.ProjectDetailId = item.ProjectDetailId;
                checkProject.ProjectType = detail.ProjectType;
                checkProject.Consumable = detail.Consumable;
                checkProject.CycleType = detail.CycleType.Value;
                checkProject.MaxValue = detail.MaxValue;
                checkProject.Method = detail.Method;
                checkProject.MinValue = detail.MinValue;
                checkProject.Part = detail.Part;
                checkProject.Standard = detail.Standard;
                checkProject.Unit = detail.Unit;
                checkProject.UseTime = detail.UseTime;
                savedData.Add(checkProject);
            }
            RF.Save(savedData);
            return true;
        }
    }
}
